sys/kern/kern_intr.c

```diff
@@ -894,6 +901,10 @@ intr_event_remove_handler(void *cookie)
 		CK_SLIST_REMOVE_PREVPTR(prevptr, ih, ih_next);
 		intr_event_barrier(ie);
 	} else {
+#ifdef HWPMC_HOOKS
+		struct trapframe *oldframe = curthread->td_intr_frame;
+		curthread->td_intr_frame = NULL;
+#endif
```

Comment:
Same issue as above.  Should `curthread->td_intr_frame` simply be left alone here?  Was there a deliberate need to simulate an empty trap frame?